prog/analysis.go

```diff
@@ -82,7 +82,7 @@ func (s *state) analyzeImpl(c *Call, resources bool) {
 					val = val[:len(val)-1]
 				}
 				switch typ.Kind {
-				case BufferString:
+				case BufferString, BufferGlob:
```

Comment:
I think it makes more sense to add BufferGlob values to s.files rather than s.strings. Then fuzzer will use the referenced files in other file operations.

Reply:
Since it's glob it's not a file but a string.
Also I cannot pass the testt for BufferFilename if I use s.files for BufferGlob.

Reply:
I've got this reply, but I can't find it on github now...

> Since it's glob it's not a file but a string.
> Also I cannot pass the testt for BufferFilename if I use s.files for BufferGlob.

The DataArg should contain actual file name we are trying to open, it's the type of the argument that contains glob pattern.
What test fails? How?

Reply:
I'm having this when use s.files

panic: sandbox escaping file name "/sys/block/loop3", s.files are map[/sys/block/l$
op3:true /sys/devices/LNXSYSTM:00:true /sys/devices/pci0000:00:true /sys/devices/u$
robe:true /sys/module/8139too:true /sys/module/msr:true /sys/power/pm_wakeup_irq:t$
ue /sys/power/wakeup_count:true]

goroutine 11 [running]:
github.com/google/syzkaller/prog.(*randGen).filename(0xc0008fc000, 0xc000060050, 0$
143e7e0, 0x0, 0x3)
        /local/mnt/workspace/jiangenj/syzkaller_qemu_setup/go/src/github.com/googl$
/syzkaller/prog/rand.go:257 +0x294


panic in 
panic(fmt.Sprintf("sandbox escaping file name %q, s.files are %v", fn, s.files))

```
func (r *randGen) filename(s *state, typ *BufferType) string {
	fn := r.filenameImpl(s)
	if fn != "" && fn[len(fn)-1] == 0 {
		panic(fmt.Sprintf("zero-terminated filename: %q", fn))
	}
	if escapingFilename(fn) {
		panic(fmt.Sprintf("sandbox escaping file name %q, s.files are %v", fn, s.files))
	}
```

I haven't enabled filaneme syscalls, but why below code called in (t *BufferType) mutate function?
	case BufferFilename:
		a.data = []byte(r.filename(s, t))

Reply:
Well, after debug, I think separate s.files is needs, let's call it s.globs